janus.c

```diff
@@ -203,7 +206,14 @@ void janus_handle_signal(int signum)
 			JANUS_PRINT("Ok, leaving immediately...\n");
 			break;
 	}
+
+	/* Wake the main loop */
+	janus_mutex_lock(&stop_mutex);
```

Comment:
Thanks for the hint, I was thinking in terms of generic event-based notifications and didn't bother actually checking how it worked with system signals. I'll fix this tomorrow.